integrations/git_test.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package integrations
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"code.gitea.io/git"
+
+	"github.com/Unknwon/com"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestClonePush_ViaHTTP_NoLogin(t *testing.T) {
+	prepareTestEnv(t)
+
+	s := http.Server{
+		Addr:    ":3000",
+		Handler: mac,
+	}
+
+	defer func() {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+		s.Shutdown(ctx)
+		cancel()
+	}()
+
+	go s.ListenAndServe()
+
+	dstPath := filepath.Join(os.TempDir(), "repo1")
+	os.RemoveAll(dstPath)
```

Comment:
`defer` ?

Reply:
I think it needs both (remove before clone - just in case it exists and defer to delete it also after)

Reply:
Not if we use `ioutil.TempDir()` :)